src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -38,9 +39,22 @@ IN_PROCESS_APPLICATION::OnAsyncCompletion(
         pHttpContext->GetResponse()->SetStatus(500, "Internal Server Error", 19, hr);
         return RQ_NOTIFICATION_FINISH_REQUEST;
     }
-
-    // Call the managed handler for async completion.
-    return m_AsyncCompletionHandler(pInProcessStoredContext->QueryManagedHttpContext(), hrCompletionStatus, cbCompletion);
+    else if (pInProcessStoredContext->QueryIsManagedRequestComplete())
+    {
+        // means PostCompletion has been called and this is the associated callback.
+        dwRequestNotificationStatus = pInProcessStoredContext->QueryAsyncCompletionStatus();
+        if (dwRequestNotificationStatus == RQ_NOTIFICATION_FINISH_REQUEST)
+        {
+            pHttpContext->DisableNotifications(RQ_END_REQUEST, 0);
```

Comment:
do we need this?

Reply:
ANCM will not receive RQ_NOTIFICATION_FINISH_REQUEST as it did not register it.
I don't think we need it. Inprocess should always call m_AsyncCompletionHandler

Reply:
Alright I'll drop it.